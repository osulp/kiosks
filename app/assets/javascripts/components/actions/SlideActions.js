export const SET_SLIDES = 'SET_SLIDES';
export const REFRESH_SLIDES = 'REFRESH_SLIDES';

export function setSlides(slides) {
  slides.push({id: 100, content: "absdg"});
  console.log(slides);
  return {
    type: SET_SLIDES,
    slides: slides
  };
}

export function refreshSlides(){
  // TODO: Ajax call to query the current set of slides?
}
