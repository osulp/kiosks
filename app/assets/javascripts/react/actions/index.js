export * from './kioskActions';
export * from './touchActions';
