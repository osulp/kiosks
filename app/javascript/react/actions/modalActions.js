export const SET_MODAL_VISIBILITY = "SET_MODAL_VISIBILITY"
export const setModalVisibility = visible => {
  return {
    type: SET_MODAL_VISIBILITY,
    visible
  }
}

export const SET_MODAL_ROOT_COMPONENT = "SET_MODAL_ROOT_COMPONENT"
export const setModalRootComponent = root_component => {
  return {
    type: SET_MODAL_ROOT_COMPONENT,
    root_component
  }
}
