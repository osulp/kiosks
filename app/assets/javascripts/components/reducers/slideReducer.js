import { SET_SLIDES, REFRESH_SLIDES } from '../actions/slideActions';

const slideReducer = (state = {}, action) => {
   switch (action.type) {
    case REFRESH_SLIDES:
      return Object.assign({}, action.slides);
    case SET_SLIDES:
      return action.slides;
    default:
      return state;
  }
}

export default slideReducer;