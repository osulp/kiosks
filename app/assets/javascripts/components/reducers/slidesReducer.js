import { SET_SLIDES, REFRESH_SLIDES } from '../actions/SlideActions';

const slidesReducer = (state = {}, action) => {
   switch (action.type) {
    case REFRESH_SLIDES:
      return Object.assign({}, action.slides);
    case SET_SLIDES:
      return action.slides;
    default:
      return state;
  }
}

export default slidesReducer;