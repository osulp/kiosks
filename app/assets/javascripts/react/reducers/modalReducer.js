import {SET_MODAL_ROOT_COMPONENT, SET_MODAL_VISIBILITY} from '../actions/modalActions';

export const initial_state = {
  visible: false,
  root_component: null
};

const modalReducer = (state = initial_state, action) => {
  switch (action.type) {
    case SET_MODAL_VISIBILITY:
      return Object.assign({}, state, { visible: action.visible });
    case SET_MODAL_ROOT_COMPONENT:
      return Object.assign({}, state, { root_component: action.root_component });
    default:
      return state;
  }
};

export default modalReducer;