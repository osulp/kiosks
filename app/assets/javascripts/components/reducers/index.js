import { combineReducers } from 'redux';
import slidesReducer from './slidesReducer';

const rootReducer = combineReducers({
  slides: slidesReducer
});

export default rootReducer;