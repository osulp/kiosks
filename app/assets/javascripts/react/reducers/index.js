import { combineReducers } from 'redux';
import kioskReducer from './kioskReducer';
import modalReducer from './modalReducer';
import touchReducer from './touchReducer';
import circReducer from './circReducer';

export const REDUCERS_SHAPE = {
  kiosk: kioskReducer,
  modal: modalReducer,
  touch: touchReducer,
  circ: circReducer
};

export default combineReducers(REDUCERS_SHAPE);
