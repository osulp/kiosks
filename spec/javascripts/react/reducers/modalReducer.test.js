import { default as reducerCreator } from "../../../../app/javascript/react/reducers/modalReducer"
import * as actionCreator from "../../../../app/javascript/react/actions/modalActions"

const empty_state = undefined

describe("Reducers::Modal", () => {
  it("matches the snapshot", () => {
    expect(reducerCreator).toMatchSnapshot()
  })
  describe("setModalRootComponent action", () => {
    it("matches the snapshot", () => {
      expect(
        reducerCreator(
          empty_state,
          actionCreator.setModalRootComponent(jest.fn())
        )
      ).toMatchSnapshot()
    })
  })
  describe("setModalVisibility action", () => {
    it("matches the snapshot", () => {
      expect(
        reducerCreator(empty_state, actionCreator.setModalVisibility(true))
      ).toMatchSnapshot()
    })
  })
})
