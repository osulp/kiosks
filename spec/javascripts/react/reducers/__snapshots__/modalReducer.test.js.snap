// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers::Modal matches the snapshot 1`] = `[Function]`;

exports[`Reducers::Modal setModalRootComponent action matches the snapshot 1`] = `
Object {
  "root_component": [MockFunction],
  "visible": false,
}
`;

exports[`Reducers::Modal setModalVisibility action matches the snapshot 1`] = `
Object {
  "root_component": null,
  "visible": true,
}
`;
